### PR TITLE
Add more filters

### PR DESCRIPTION
**Pull request by rvl**
I found that my app needed character filters and the "stemmer" and "stop" token filters, so here they are.

Cheers!

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+UNRELEASED
+========
+- @rvl
+  - Add [character filters][] to the `AnalyzerDefinition` type.
+  - Add the "stemmer" and "stop" [token filters][] to `TokenFilterDefinition`.
+
+[Character Filters]: https://www.elastic.co/guide/en/elasticsearch/reference/5.6/analysis-charfilters.html
+[Token Filters]: https://www.elastic.co/guide/en/elasticsearch/reference/5.6/analysis-tokenfilters.html
```
Comment (bitemyapp) [inline changelog.md]: ty for taking care of this for me

**Comment by bitemyapp** [comment]
LG2M.

I'll integrate this work into the ongoing PR at https://github.com/bitemyapp/bloodhound/pull/223 manually.

**Comment by rvl** [comment]
Sounds good, thanks.

File: src/Database/V5/Bloodhound/Types.hs [hunk 2]
```diff
@@ -1394,7 +1460,7 @@ newtype TypeName =
 newtype PercentMatch =
   PercentMatch Double deriving (Eq, Read, Show, Generic, ToJSON, FromJSON, Typeable)
 newtype StopWord =
-  StopWord Text deriving (Eq, Read, Show, Generic, ToJSON, FromJSON, Typeable)
+  StopWord Text deriving (Eq, Read, Show, Generic, ToJSON, FromJSON, Typeable, IsString)
```
Comment (bitemyapp) [inline src/Database/V5/Bloodhound/Types.hs]: Don't add `IsString` instances to newtypes, it defeats the purpose of the newtype.

**Comment by bitemyapp** [comment]
Fret not, this is actually merged, but I do so manually into `master`.

Thank you!

**Pull request closed without merging**